baseplate/frameworks/pyramid/__init__.py

```diff
@@ -68,11 +75,60 @@ def close(self) -> None:
             self.app_iter.close()  # type: ignore
 
 
+PROM_NAMESPACE = "http_server"
+
+HISTOGRAM_LABELS = [
+    "http_method",
+    "http_endpoint",
+    "http_success",
+]
+REQUEST_LATENCY = Histogram(
+    f"{PROM_NAMESPACE}_latency_seconds",
+    "Time spent processing requests",
+    HISTOGRAM_LABELS,
+    buckets=default_latency_buckets,
+)
+REQUEST_SIZE = Histogram(
+    f"{PROM_NAMESPACE}_request_size_bytes",
+    "Size of incoming requests in bytes",
+    HISTOGRAM_LABELS,
+    buckets=default_size_buckets,
+)
+RESPONSE_SIZE = Histogram(
+    f"{PROM_NAMESPACE}_response_size_bytes",
+    "Size of outgoing responses in bytes",
+    HISTOGRAM_LABELS,
+    buckets=default_size_buckets,
+)
+REQUESTS_TOTAL = Counter(
+    f"{PROM_NAMESPACE}_requests_total",
+    "Total number of request handled",
+    [
+        *HISTOGRAM_LABELS,
+        "http_response_code",
+    ],
+)
+ACTIVE_REQUESTS = Gauge(
+    f"{PROM_NAMESPACE}_active_requests",
+    "Current requests in flight",
+    [
+        "http_method",
+        "http_endpoint",
+    ],
+)
+
+
 def _make_baseplate_tween(
     handler: Callable[[Request], Response], _registry: Registry
 ) -> Callable[[Request], Response]:
     def baseplate_tween(request: Request) -> Response:
+        time_started = time.perf_counter()
+
+        response: Optional[Response] = None
+        endpoint = request.matched_route.pattern if request.matched_route else ""
```

Comment:
🔕 I'm making a mental note that we should probably come back to this, since a blank label and a label that says "404" might be a helpful differentiator (beyond success=false and the response code). That said this is already helpful for figuring out 404s on matched routes (eg the route decided to respond 404) vs totally random URL requests.